Sources/StreamChat/WebSocketClient/Events/ChannelEvents_Tests.swift

```diff
@@ -64,10 +65,274 @@ class ChannelEvents_Tests: XCTestCase {
+        let event = try eventDecoder.decode(from: json) as? NotificationChannelMutesUpdatedEvent
+        
+        try client.databaseContainer.createCurrentUser(id: "luke_skywalker")
+        
+        // Create mute payloads for current user so there are some muted Channels:
+        let mutePayloads: [MutedChannelPayload<NoExtraData>] = [
+            .init(
+                mutedChannel: .dummy(cid: .unique),
+                user: dummyUser(id: "luke_skywalker"),
+                createdAt: .unique,
+                updatedAt: .unique
+            ),
+            .init(
+                mutedChannel: .dummy(cid: .unique),
+                user: dummyUser(id: "luke_skywalker"),
+                createdAt: .unique,
+                updatedAt: .unique
+            )
+        ]
+
+        try client.databaseContainer.writeSynchronously { session in
+            // Save channel mutes to database.
+            for payload in mutePayloads {
+                try session.saveChannelMute(payload: payload)
+            }
+        }
+        
+        let unwrappedUser = try XCTUnwrap(client.databaseContainer.viewContext.currentUser)
+        XCTAssertFalse(unwrappedUser.user.channelMutes.isEmpty)
+                
+        let unwrappedEvent = try XCTUnwrap(event)
+        client.eventNotificationCenter.process(unwrappedEvent)
+        
+        AssertAsync {
+            Assert.willBeTrue(self.client.databaseContainer.viewContext.currentUser?.user.channelMutes.isEmpty ?? false)
+        }
+    }
+    
+    func test_NotificationChannelMutesUpdatedWithSomeMutesEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "NotificationChannelMutesUpdatedWithSomeMutedChannels")
+        let event = try eventDecoder.decode(from: json) as? NotificationChannelMutesUpdatedEvent
+        
+        try client.databaseContainer.createCurrentUser(id: "luke_skywalker")
+        
+        let unwrappedUser = try XCTUnwrap(client.databaseContainer.viewContext.currentUser)
+        XCTAssertTrue(unwrappedUser.user.channelMutes.isEmpty)
+                
+        let unwrappedEvent = try XCTUnwrap(event)
+        client.eventNotificationCenter.process(unwrappedEvent)
+        
+        AssertAsync {
+            Assert.willBeFalse(self.client.databaseContainer.viewContext.currentUser?.user.channelMutes.isEmpty ?? true)
+        }
+    }
+    
+    func test_NotificationMarkAllReadEventPayload_isHandled() throws {
+        let json = XCTestCase.mockData(fromFile: "NotificationMarkRead")
+        let event = try eventDecoder.decode(from: json) as? NotificationMarkReadEvent
+        
+        let channelId: ChannelId = .init(type: .messaging, id: "general")
+        let unwrappedEvent = try XCTUnwrap(event)
+        
+        // For message to be received, we need to have channel:
+        try client.databaseContainer.createChannel(
+            cid: channelId,
+            withMessages: true,
+            withQuery: false
+        )
```

Comment:
aaaah, you probably meant `For the EVENT to be received ...`?

Reply:
Done.